### ISSUE TITLE
Mention turning off range checks in compile doc? 

### ISSUE THREAD

**Issue by jgabry** [stan-dev/cmdstanr#506]
In the upcoming release we have the option of turning off range checks to improve performance. Should we explicitly mention this option in the compile method doc?

**Comment by rok-cesnovar** [comment]
Yes. And at this point there are now enough tips and tricks for performance (for the last few % of gains) where we could make it a vignette maybe?

**Comment by jgabry** [comment]
> we could make it a vignette maybe

Good idea!

**Comment by jgabry** [comment]
Going to update the title of this issue to reflect that it should include other performance tips and tricks. Possibly in the compile doc or possibly in its own vignette. @avehtari has a blog post that demonstrates other performance tips that could be included. @avehtari has a blog post that demonstrates other performance tips that could be included.

**Comment by jgabry** [comment]
A vignette about performance tips and tricks could be broader than just related to compilation and could include things like #1073. And @ssp3nc3r maybe you have other tips that could be included too? I know you've spent a decent amount of time investigating ways to improve performance.

**Comment by ssp3nc3r** [comment]
Other than the above suggestion, and obviously following the general Stan advice to best parameterize the model itself, I've been finding the largest speedups beyond parameterization alone by moving from Stan's AD tape to using C++ analytic gradients and more recently Metal on the GPU. But those aren't cmdstanr specific beyond it exposing some useful tools that let me validate my C++ code against Stan's AD tape, which is motivation for the subject of a different pull request in progress.

**Comment by ssp3nc3r** [comment]
There is a lot of useful tips in the Stan User Guide, like indexing efficiency: https://mc-stan.org/docs/stan-users-guide/matrices-arrays.html#indexing-efficiency.section and https://mc-stan.org/docs/stan-users-guide/efficiency-tuning.html ... but those do not really have worked out case studies or examples ... would the vignette about performance be more about implementing some of these?
